### ISSUE TITLE
Windows URI Scheme not working in CefSharp

### ISSUE THREAD

**Issue by rover886** [cefsharp/CefSharp#1442]
I am referencing CefSharp version 43.

Using CefSharp I have developed an flavoring application, which is embed with a intranet URL. Our network admins do register some [custom URI scheme(s)](https://msdn.microsoft.com/en-us/library/aa767914%28v=vs.85%29.aspx) in windows registry and installs respected application for that scheme through domain control . This URI scheme will get activated when URL from web is like `orgerp://empid=100`, windows will look for URI scheme and opens the respected application for which it is registered.

If employees are using CefSharp based flavoring application then these URI schemes not get activated. I know in CEF we can register for scheme through `settings.RegisterScheme`, but this will not give freedom to network admins. Every time time if they want to launch any new URI scheme then each time we need to build the flavoring application with new scheme. 

If an protocol/scheme is not registered with `CEF` then it should process normally so if that scheme registered in windows registry will get activated, like other browser (chrome) does, if we enter `chrome://settings/` then browser's settings page get open if we enter `orgerp://empid=100` then it activates windows schemes.

Is I am missing something or this is only the behavior of `CEF`?

**Comment by amaitland** [comment]
If I understand you correctly then you need to implement `IRequestHandler.OnProtocolExecution`, return true for those `URI's` you wish to be passed to the `OS` for execution.

https://github.com/cefsharp/CefSharp/blob/cefsharp/43/CefSharp/IRequestHandler.cs#L156

**Comment by rover886** [comment]
Ohh yes, how can I forget this handler :P, Anyways thanks @amaitland. :+1: 

But, as you said I returned true from `IRequestHandler.OnProtocolExecution` then the custom URL (`orgerp://empid=100`) for which request has activated and respected application is get executed, but in browser's address bar (I am using CefSharp.WinForms.Example project to test this, in my application no address bar is used) the custom URI get displayed and white page has come. So to avoid this I written below lines in `IRequestHandler.OnBeforeBrowse`

```
if(request.Url.StartsWith("orgerp") || request.Url.StartsWith("orglms"))
            {
                return true;
            }
            return false;
```
which is again the same thing I have to build the application for every schemes otherwise white page will get displayed. I also tried using `browser.StopLoad();` before `return true` inside  `IRequestHandler.OnProtocolExecution`.
